File: check_css.py

with open('client/src/index.css', 'r') as f:
    content = f.read()

stack = []
for i, char in enumerate(content):
    if char == '{':
        stack.append(i)
    elif char == '}':
        if not stack:
            print(f"Extra closing brace at position {i}")
        else:
            stack.pop()

if stack:
    for pos in stack:
        print(f"Unclosed opening brace starting at position {pos}")
else:
    print("All braces are matched.")

File: check_css.js

const fs = require('fs');
const content = fs.readFileSync('client/src/index.css', 'utf8');

let stack = [];
for (let i = 0; i < content.length; i++) {
    const char = content[i];
    if (char === '{') {
        stack.push(i);
    } else if (char === '}') {
        if (stack.length === 0) {
            console.log(`Extra closing brace at position ${i}`);
        } else {
            stack.pop();
        }
    }
}

if (stack.length > 0) {
    stack.forEach(pos => {
        console.log(`Unclosed opening brace starting at position ${pos}`);
    });
} else {
    console.log("All braces are matched.");
}
